### PR TITLE
[WW-5143] [WW-5142] Upgrades OVal to ver. 3.2.1 and upgrades XStream to ver 1.4.18

### PR DESCRIPTION
**Pull request by lukaszlenart**
Upgrades XStream and OVal
[WW-5142](https://issues.apache.org/jira/browse/WW-5142)
[WW-5143](https://issues.apache.org/jira/browse/WW-5143)

### CODE REVIEW

File: plugins/oval/src/test/java/org/apache/struts2/oval/interceptor/OValValidationInterceptorTest.java [hunk 1]
```diff
@@ -308,24 +296,12 @@ public void testMemberObject() throws Exception {
...
-            // Oval version <= 1.40 validation error for minimum length reports: "street cannot be smaller than 7 characters".
-            assertValue(fieldErrors, "person.address.street", Arrays.asList("street cannot be smaller than 7 characters"));
-        } catch (AssertionFailedError afe) {
-            // Oval version >= 1.50 validation error for invalid data reports: "address is invalid".
-            assertValue(fieldErrors, "person.address", Arrays.asList("address is invalid"));
-            // Oval version >= 1.50 validation error for minimum length reports: "street cannot be shorter than 7 characters".
-            assertValue(fieldErrors, "person.address.street", Arrays.asList("street cannot be shorter than 7 characters"));
-        }
-
+        assertEquals(3, fieldErrors.size()); // 5: as there will be field errors for 'person' and 'person.address' themselves
```
Comment (yasserzamani) [inline plugins/oval/src/test/java/org/apache/struts2/oval/interceptor/OValValidationInterceptorTest.java:299]: nitpick: // 5: in comment should be // 3:

File: plugins/oval/src/test/java/org/apache/struts2/oval/interceptor/OValValidationInterceptorTest.java [hunk 2]
```diff
@@ -281,21 +280,10 @@ public void testModelDrivenAction() throws Exception {
...
-            // Oval version <= 1.40 validation error for minimum length reports: "street cannot be smaller than 7 characters".
-            assertValue(fieldErrors, "address.street", Arrays.asList("street cannot be smaller than 7 characters"));
-        } catch (AssertionFailedError afe) {
-            // Oval version >= 1.50 validation error for invalid data reports: "address is invalid".
-            assertValue(fieldErrors, "address", Arrays.asList("address is invalid"));
-            // Oval version >= 1.50 validation error for minimum length reports: "street cannot be shorter than 7 characters".
-            assertValue(fieldErrors, "address.street", Arrays.asList("street cannot be shorter than 7 characters"));
-        }
-
+        assertEquals(3, fieldErrors.size()); // 5: as there will be field errors for 'model' and 'address' themselves
```
Comment (yasserzamani) [inline plugins/oval/src/test/java/org/apache/struts2/oval/interceptor/OValValidationInterceptorTest.java:283]: nitpick: // 5: in comment should be // 3:

File: plugins/oval/src/test/java/org/apache/struts2/oval/interceptor/OValValidationInterceptorTest.java (hunk 1 above)
Comment (lukaszlenart) [reply plugins/oval/src/test/java/org/apache/struts2/oval/interceptor/OValValidationInterceptorTest.java]: 👍

File: plugins/oval/src/test/java/org/apache/struts2/oval/interceptor/OValValidationInterceptorTest.java (hunk 2 above)
Comment (lukaszlenart) [reply plugins/oval/src/test/java/org/apache/struts2/oval/interceptor/OValValidationInterceptorTest.java]: 👍

**Comment by yasserzamani** [comment]
Frankly I didn't completely understand all changes but tests show that it's operating as expected as before. LGTM :+1:

**Pull request merged**